.github/workflows/release.yaml

```diff
@@ -21,4 +21,4 @@ jobs:
         Name-Email: github-action@etcd.io
         Expire-Date: 0
         EOF
-        DRY_RUN=true BRANCH=main ./scripts/release.sh --no-upload --no-docker-push 3.6.99
+        DRY_RUN=true ./scripts/release.sh --no-upload --no-docker-push --current-branch 3.6.99
```

Comment:
This sounds more like a `--local` flag.

Reply:
This seems not a big deal, I'd like to keep it as it's. `--current-branch ` seems a little better.  A current branch isn't necessary a local branch. github runner already cloned the repo and checked out the branch, we just kick off the release process in current branch which the github runner already checked out.

Reply:
I'm against writing scripts that work only for github runner. As a maintainer I want to be able to run the same script locally to be able to debug the release script. First phase of debugging release script should not be figuring out differences between local and github runner setup. 

Please don't put any assumptions about running in github runner.

Reply:
To understand both options: 
  1. `--local` would mean: do not clone, keep using 'release-3.6.46'  branch.  
  2. `--current_branch` overrides both BRANCH name and forces usage of local repository. 

So the question is whether there is a valid use-case when someone has a git-cloned repository and want's to build an already tagged release. And to me it makes sense - for a case someone does not trust etcd-team provided binaries. 

So the less interactions between flags the better. 

To accommodate the github usecase the invocation would look like: 

```
BRANCH=$(git symbolic-ref --short HEAD) DRY_RUN=true ./scripts/release.sh --no-upload --no-docker-push --local 3.6.99
```

I would also consider `--in-place` flag, as `--local` might suggests there is an option to perform the build on a remote machine.

Reply:
Updated.

Reply:
thx